### PR TITLE
Enforce string-formatting with f-strings

### PR DESCRIPTION
**Pull request by sgugger**
# What does this PR do?

This PR removes any strings formatted with `.format` or `%` to use f-strings exclusively (unless there is a very good reason to use the other syntax, or the file is in a research_project/legacy folder).

The mix of three syntaxes does not make any sense and we all agree in the team that f-strings are more readable. Now that Python 3.5 is officially dead, there is no reason not to switch fully to f-strings.

cc @stas00  as we had a conversation about that.

### CODE REVIEW

**Review by stas00 — APPROVED** [review]
You're a machine, @sgugger! So much editing effort! 

This is amazing! 

Thank you!

File: src/transformers/models/t5/modeling_t5.py [hunk 1]
```diff
@@ -888,9 +886,7 @@ def forward(
         mask_seq_length = past_key_values[0][0].shape[2] + seq_length if past_key_values is not None else seq_length
 
         if use_cache is True:
-            assert self.is_decoder, ":obj:`use_cache` can only be set to `True` if {} is used as a decoder".format(
-                self
-            )
+            assert self.is_decoder, ":obj:`use_cache` can only be set to `True` if {self} is used as a decoder"
```
Comment (LysandreJik) [inline src/transformers/models/t5/modeling_t5.py:889]:
```suggestion
            assert self.is_decoder, f":obj:`use_cache` can only be set to `True` if {self} is used as a decoder"
```

File: src/transformers/models/wav2vec2/tokenization_wav2vec2.py [hunk 2]
```diff
@@ -514,7 +514,7 @@ def _decode(
 
     def save_vocabulary(self, save_directory: str, filename_prefix: Optional[str] = None) -> Tuple[str]:
         if not os.path.isdir(save_directory):
-            logger.error("Vocabulary path ({}) should be a directory".format(save_directory))
+            logger.error("Vocabulary path ({save_directory}) should be a directory")
```
Comment (LysandreJik) [inline src/transformers/models/wav2vec2/tokenization_wav2vec2.py:517]:
```suggestion
            logger.error(f"Vocabulary path ({save_directory}) should be a directory")
```

**Review by LysandreJik — APPROVED** [review]
LGTM! Fantastic work @sgugger, thanks so much for taking care of this!

**Pull request merged**